components/dashboard/src/settings/ProfileInformation.tsx

```diff
@@ -0,0 +1,208 @@
+            return "Please enter a valid email.";
+        }
+        return "";
+    }
+
+    export function NudgeForProfileUpdateModal() {
+        const { user, setUser } = useContext(UserContext);
+        const original = ProfileState.getProfileState(user!);
+        const [profileState, setProfileState] = useState(original);
+        const [errorMessage, setErrorMessage] = useState("");
+        const [visible, setVisible] = useState(shouldNudgeForUpdate(user!));
+
+        const saveProfileState = () => {
+            const error = ProfileState.validate(profileState);
+            setErrorMessage(error);
+            if (error) {
+                return;
+            }
+            const updatedUser = ProfileState.setProfileState(user!, profileState);
+            setUser(updatedUser);
+            getGitpodService().server.updateLoggedInUser(updatedUser);
+            setVisible(shouldNudgeForUpdate(updatedUser!));
+        };
+
+        const cancelProfileUpdate = () => {
+            setProfileState(original);
+            saveProfileState();
+        };
+
+        return (
+            <Modal
+                title="Update Profile Information"
+                visible={visible}
+                onClose={cancelProfileUpdate}
+                closeable={true}
+                className="_max-w-xl"
+                buttons={
+                    <div>
+                        <button className="secondary" onClick={cancelProfileUpdate}>
+                            Dismiss
+                        </button>
+                        <button className="ml-4" onClick={saveProfileState}>
+                            Update Profile
+                        </button>
+                    </div>
+                }
+            >
+                <ProfileInformation
+                    profileState={profileState}
+                    errorMessage={errorMessage}
+                    updated={false}
+                    setProfileState={setProfileState}
+                />
+            </Modal>
+        );
+    }
+}
+
+export default function ProfileInformation(props: {
+    profileState: ProfileState.ProfileState;
+    setProfileState: (newState: ProfileState.ProfileState) => void;
+    errorMessage: string;
+    updated: boolean;
+    children?: React.ReactChild[] | React.ReactChild;
+}) {
+    return (
+        <div>
+            <p className="text-base text-gray-500 pb-4 max-w-xl">
+                The following information will be used to set up Git configuration. You can override Git author name and
+                email per project by using the default environment variables <CodeText>GIT_AUTHOR_NAME</CodeText>,{" "}
+                <CodeText>GIT_COMMITTER_NAME</CodeText>, <CodeText>GIT_AUTHOR_EMAIL</CodeText> and{" "}
+                <CodeText>GIT_COMMITTER_EMAIL</CodeText>.
+            </p>
+            {props.errorMessage.length > 0 && (
+                <Alert type="error" closable={true} className="mb-2 max-w-xl rounded-md">
```

Comment:
**issue(non-blocking)**: The closable icon is not aligned vertically in the middle but that's out of the scope of this PR. Good news is that now we can fix once and update all closable alert component instances.

Reply:
Do you want this to be always centered even when the contents is longer?
<img width="614" alt="Screenshot 2022-07-01 at 12 36 47" src="https://user-images.githubusercontent.com/372735/176878888-ed0c002f-64d6-4bdd-8d61-f00c7110c1d2.png">

Reply:
Aligning these to the top would be great. 🔝 

Also, ideally, both the icon on the left, and the closable button on the right use the same bounding box (icon live area + padding) height (`20px`) as the line height of the paragraph text in the middle so that the first line always align with the two icons horizontally, see screenshot below.

<img width="897" alt="alert" src="https://user-images.githubusercontent.com/120486/176881116-087d69be-1d50-4cf7-a407-bc9b0cf75243.png">

Reply:
Ok, the reason the x was a bitz off is that we used a h-3 instead of h-4. I've fixed this now.

Reply:
Perfect! One last point that could be part of a follow-up issue or PR is not hide the alert after updating the profile when the user is updating any of the values in the inputs below, just like it's done with the error alert component. The alert should be hidden only when the user explicitly closes (dismiss) the alert or moves to another page away from **/account**. Other than that, this looks ready-to-merge! 🛹 

https://user-images.githubusercontent.com/120486/176913316-17d17d34-f077-4158-b324-d94fb6fae789.mov

Reply:
Fair point—Let's keep it as is. 🤝